moveit_core/constraint_samplers/test/pr2_arm_kinematics_plugin.cpp

```diff
@@ -300,46 +297,48 @@ bool PR2ArmKinematicsPlugin::initialize(const moveit::core::RobotModel& robot_mo
     {
       for (unsigned int i = 0; i < ik_solver_info_.joint_names.size(); i++)
       {
-        ROS_DEBUG_NAMED("pr2_arm_kinematics_plugin", "PR2Kinematics:: joint name: %s",
+        RCLCPP_WARN(LOGGER_PR2_ARM_KINEMATICS_PLUGIN, "PR2Kinematics:: joint name: %s",
                         ik_solver_info_.joint_names[i].c_str());
       }
       for (unsigned int i = 0; i < ik_solver_info_.link_names.size(); i++)
       {
-        ROS_DEBUG_NAMED("pr2_arm_kinematics_plugin", "PR2Kinematics can solve IK for %s",
+        RCLCPP_WARN(LOGGER_PR2_ARM_KINEMATICS_PLUGIN, "PR2Kinematics can solve IK for %s",
                         ik_solver_info_.link_names[i].c_str());
       }
       for (unsigned int i = 0; i < fk_solver_info_.link_names.size(); i++)
       {
-        ROS_DEBUG_NAMED("pr2_arm_kinematics_plugin", "PR2Kinematics can solve FK for %s",
+        RCLCPP_WARN(LOGGER_PR2_ARM_KINEMATICS_PLUGIN, "PR2Kinematics can solve FK for %s",
                         fk_solver_info_.link_names[i].c_str());
       }
-      ROS_DEBUG_NAMED("pr2_arm_kinematics_plugin", "PR2KinematicsPlugin::active for %s", group_name.c_str());
+      RCLCPP_WARN(LOGGER_PR2_ARM_KINEMATICS_PLUGIN, "PR2KinematicsPlugin::active for %s", group_name.c_str());
     }
     active_ = true;
   }
   return active_;
 }
 
-bool PR2ArmKinematicsPlugin::getPositionIK(const geometry_msgs::Pose& ik_pose, const std::vector<double>& ik_seed_state,
+bool PR2ArmKinematicsPlugin::getPositionIK(const geometry_msgs::msg::Pose& ik_pose, const std::vector<double>& ik_seed_state,
                                            std::vector<double>& solution, moveit_msgs::msg::MoveItErrorCodes& error_code,
                                            const kinematics::KinematicsQueryOptions& options) const
 {
   return false;
 }
 
-bool PR2ArmKinematicsPlugin::searchPositionIK(const geometry_msgs::Pose& ik_pose,
+bool PR2ArmKinematicsPlugin::searchPositionIK(const geometry_msgs::msg::Pose& ik_pose,
                                               const std::vector<double>& ik_seed_state, double timeout,
                                               std::vector<double>& solution, moveit_msgs::msg::MoveItErrorCodes& error_code,
                                               const kinematics::KinematicsQueryOptions& options) const
 {
   if (!active_)
   {
-    ROS_ERROR("kinematics not active");
+    RCLCPP_ERROR(LOGGER_PR2_ARM_KINEMATICS_PLUGIN,"kinematics not active");
     error_code.val = error_code.PLANNING_FAILED;
     return false;
   }
-  KDL::Frame pose_desired;
-  tf2::fromMsg(ik_pose, pose_desired);
+  KDL::Frame pose_desired = KDL::Frame(KDL::Rotation::Quaternion(ik_pose.orientation.x, ik_pose.orientation.y,
```

Comment:
This should use fromMsg() from tf2_kdl.h once geometry2 is ported to ROS2. Please add a TODO

Reply:
There is actually an implementation for such transformation on 
https://github.com/ros2/geometry2/blob/234725f30ade83caa4bcf019ee854199aa31bef5/tf2_kdl/include/tf2_kdl/tf2_kdl.h#L273-L282, you have to pass a PoseStamped instead of a pose, so I suggest the following changes.

```suggestion
  geometry_msgs::msg::PoseStamped ik_pose_stamped;
  ik_pose_stamped.pose = ik_pose;

  tf2::Stamped<KDL::Frame> pose_desired;

  tf2::fromMsg(ik_pose_stamped, pose_desired);
```

Reply:
looks good to me

Reply:
@anasarrak I'm going to merge this but I would like to see this in a follow up PR